BitwardenShared/UI/Platform/Application/AppProcessor.swift

```diff
@@ -380,6 +371,47 @@ extension AppProcessor {
         }
     }
 
+    /// Handle a "bitwarden://" url.
+    ///
+    /// - Parameter url: The Bitwarden URL received by the app.
+    ///
+    private func handleBitwardenUrl(url: URL) async {
+        guard let components = URLComponents(url: url, resolvingAgainstBaseURL: true) else { return }
+
+        if components.host == "settings", components.path == "/account_security" {
+            let accountSecurityRoute = AppRoute.tab(.settings(.accountSecurity))
+            guard let userId = try? await services.stateService.getActiveAccountId(),
+                  !services.vaultTimeoutService.isLocked(userId: userId),
+                  await (try? services.vaultTimeoutService.hasPassedSessionTimeout(userId: userId)) == false
+            else {
+                await coordinator?.handleEvent(.setAuthCompletionRoute(accountSecurityRoute))
+                return
+            }
+            coordinator?.navigate(to: accountSecurityRoute)
```

Comment:
🎨 This is doing the same as in `handleOtpAuthUrl`, could the shared part be extracted to a method where we just pass the `AppRoute` and it takes care of the locked logic and navigation using such `AppRoute`?

Reply:
Refactored in the latest. Now the `handle*Url` methods return `AppRoute?` and we just go through one path to handle whether we need to wait for unlock or not.  👍